internal/dataplane/parser/translate_ingress_test.go

```diff
@@ -1102,6 +1102,195 @@ func TestFromIngressV1(t *testing.T) {
 			assert.Equal(t, *parsedInfo.ServiceNameToServices["foo-namespace.foo-2.foo-svc.80"].Host, "foo-svc.foo-namespace.80.svc")
 		})
 	})
+
+	t.Run("CombinedRoutes=on && CombinedServices=on", func(t *testing.T) {
+		setupParser := func(t *testing.T, store store.Storer) *Parser {
```

Comment:
can we use a table driven test structure here instead? This could make the file shorter. (maybe we could do this in a separate refactoring PR)

Reply:
Yes, I agree that would be useful 👍 Let's tackle it separately: https://github.com/Kong/kubernetes-ingress-controller/issues/3978